app/src/main/java/models/RoomPostWithObjects.java

```diff
@@ -0,0 +1,35 @@
+package models;
+
+import androidx.room.Embedded;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RoomPostWithObjects {
+    @Embedded(prefix="user_")
+    RoomUser roomUser;
+
+    @Embedded
+    RoomPost roomPost;
+
+    public static List<RoomPost> getRoomPostList(List<RoomPostWithObjects> postsWithObjects, Group group) {
+        List<RoomPost> posts = new ArrayList<>();
+        for (int i = 0; i < postsWithObjects.size(); i++) {
+            RoomPost post = postsWithObjects.get(i).roomPost;
+            post.roomUser = postsWithObjects.get(i).roomUser;
+            if (post.roomGroupId.equals(group.getObjectId())) {
```

Comment:
nit: check equality first, if it doesnt match, just continue the loop instead of creating a post/setting roomuser attribute first